src/dxvk/dxvk_instance.cpp

```diff
@@ -232,6 +232,15 @@ namespace dxvk {
       Logger::warn("DXVK: No adapters found. Please check your "
                    "device filter settings and Vulkan setup.");
     }
+    else if (result[0]->deviceProperties().deviceType == VK_PHYSICAL_DEVICE_TYPE_DISCRETE_GPU) {
+      for (uint32_t i = 1; i < result.size(); i++) {
+        if (result[i]->deviceProperties().deviceType == VK_PHYSICAL_DEVICE_TYPE_INTEGRATED_GPU)
+        {
+          result[i]->linkToDGPU(result[0]);
```

Comment:
what's the rationale behind any of this? what if there's more than one dedicated GPU present on a system?

Reply:
That's how discrete GPU reporting works on Windows, as far as my testing went. QueryDisplayConfig always reports the actual connection. That is, if a monitor is plugged to integrated GPU it will be always reported as such through QueryDisplayConfig. dxgi outputs to adapters binding however depends on whether discrete or integrated GPU is selected for the app (auto, or forced by "Powersave" settings I was changing for testing this). If discrete GPU is selected for the game the outputs will be reported from discrete GPU adapter. DXVK always tries to simulate the case when discrete GPU is selected, so to match Windows and not to confuse some games the outputs should be linked to discrete one.

I am afraid the case when there are multiple drivers for the same GPU device might be already broken for those games which are sensitive to correct adapter / displays configuration reporting. To have anything about that right we must end up with an adapter being configured in winex11.drv for it. If this part went well and we have an adapter this part here should probably be fine in general (we will have outputs linked to some of those), although I have no idea what is the "correct" adapter in this case, not sure if we have any good way to guess. That hopefully won't break the things completely, it will bind to one of, and if that will result in the game using not the adapter / driver user wants to use, the only way seems to be to choose / select just one adapter on host, or change their order somehow.

Can you maybe describe how exactly I can simulate such a configuration to see how it will look like?

Reply:
PS Maybe it is actually best to just skip this redirection (at least for now) if more than one dGPU is found, for whatever reason (multiple drivers or real GPUs), and fallback to reporting displays bound as reported by QueryDisplayConfig. As it is indeed unclear what to do in this case, and I don't even have such a configuration with real GPUs to test, and that is apparently not a dual-GPU laptop config (or something special with multiple drivers where things either happen to work or user will have to disable one of those).